toolkit/tools/pkg/profile/profile.go

```diff
@@ -0,0 +1,76 @@
+package profile
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"runtime/pprof"
+	"runtime/trace"
+)
+
+type Profiler struct {
+	cpuProfFile *os.File
+	memProfFile *os.File
+	traceFile   *os.File
+}
+
+func StartProfiling(cpuProfFile, memProfFile, traceFile string, enableCpuProf, enableMemProf, enableTrace bool) (*Profiler, error) {
```

Comment:
We could clean up the signature here even more, since we are already creating this container  when we call `SetupProfileFlags()`
```suggestion
func StartProfiling(profileFlags exe.ProfileFlags) (*Profiler, error) {
```